Unable to run jnlp files for Intel MFSys25 KVM (Could not read or parse the JNLP File)

When using javaws (icedtea-netx-1.8-0ubuntu8, on Xubuntu 19.10) to run a jnlp file generated by a Intel MFSys25 V1 (latest eol firmware), It fails due to 'Could not read or parse the JNLP File'.

I believe due to the following line at the end of the .jnlp file
<?-- JViewerVersion 3.30a -->
(Specifically deleting that line allows it to work.  This works out of the box in Oracle's Java version 8

JNLP file is attacked with some information masked/removed.


Stacktrace:

net.sourceforge.jnlp.LaunchException: Fatal: Read Error: Could not read or parse the JNLP file. You can try to download this file manually and send it as bug report to IcedTea-Web team.     at net.sourceforge.jnlp.Launcher.fromUrl(Launcher.java:489)     at net.sourceforge.jnlp.Launcher.launch(Launcher.java:289)     at net.sourceforge.jnlp.runtime.JnlpBoot.run(JnlpBoot.java:67)     at net.sourceforge.jnlp.runtime.Boot.run(Boot.java:270)     at net.sourceforge.jnlp.runtime.Boot.run(Boot.java:63)     at java.base/java.security.AccessController.doPrivileged(Native Method)     at net.sourceforge.jnlp.runtime.Boot.main(Boot.java:210) Caused by: net.sourceforge.jnlp.ParseException: Invalid XML document syntax.     at net.sourceforge.jnlp.XMLParser.getRootNode(XmlParser.java:118)     at net.sourceforge.jnlp.MalformedXMLParser.getRootNode(MalformedXMLParser.java:79)     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)     at java.base/java.lang.reflect.Method.invoke(Method.java:566)     at net.sourceforge.jnlp.Parser.getRootNode(Parser.java:1380)     at net.sourceforge.jnlp.JNLPFile.parse(JNLPFile.java:866)     at net.sourceforge.jnlp.JNLPFile.<init>(JNLPFile.java:233)     at net.sourceforge.jnlp.JNLPFile.<init>(JNLPFile.java:215)     at net.sourceforge.jnlp.JNLPFile.<init>(JNLPFile.java:200)     at net.sourceforge.jnlp.JNLPFile.<init>(JNLPFile.java:186)     at net.sourceforge.jnlp.Launcher.fromUrl(Launcher.java:453)     ... 6 more Caused by: net.sourceforge.nanoxml.XMLParseException: XML Parse Exception during parsing of a jnlp element at line 80: Expected: =     at net.sourceforge.nanoxml.XMLElement.expectedInput(XMLElement.java:1184)     at net.sourceforge.nanoxml.XMLElement.scanElement(XMLElement.java:962)     at net.sourceforge.nanoxml.XMLElement.scanElement(XMLElement.java:1018)     at net.sourceforge.nanoxml.XMLElement.parseFromReader(XMLElement.java:514)     at net.sourceforge.nanoxml.XMLElement.parseFromReader(XMLElement.java:466)     at net.sourceforge.jnlp.XMLParser.getRootNode(XmlParser.java:114)     ... 18 more 
netx: Read Error: Could not read or parse the JNLP file. (Invalid XML document syntax. (XML Parse Exception during parsing of a jnlp element at line 80: Expected: =))
line: 83

line: 82
line: 81 </jnlp>
line: 80 <?_-- JViewerVersion 3.30a --?>
line: 79
line: 78  </application-desc>
line: 77    <argument>1</argument>
line: 76    <argument>-keyboardtype</argument>
line: 75    <argument>2</argument>
line: 74    <argument>-mousemode</argument>
line: 73    <argument>4</argument>
line: 72    <argument>-accelthreshold</argument>
line: 71    <argument>2</argument>
line: 70    <argument>-mouseaccel</argument>
line: 69    <argument>6371</argument>
line: 68    <argument>-fdport</argument>
line: 67    <argument>6341</argument>
line: 66    <argument>-cdport</argument>
line: 65    <argument>0</argument>
line: 64    <argument>-sslvm</argument>
line: 63    <argument>0</argument>
line: 62    <argument>-sslkvm</argument>
line: 61    <argument>31313732303136303331303034303132333435363738393031323334353637383930313233343536373839303132333435363738393031323334353637383930</argument>
line: 60    <argument>-ticket</argument>
line: 59    <argument>6241</argument>
line: 58    <argument>-vport</argument>
line: 57    <argument>intelunit2.vc.bravenet.com</argument>
line: 56    <argument>-hostname</argument>
line: 55    <argument>Remote KVM - Server 1</argument>
line: 54    <argument>-fullscreentitle</argument>
line: 53    <argument>Remote KVM Session to Server 1</argument>
line: 52    <argument>-title</argument>
line: 51  <application-desc>
line: 50  </resources>
line: 49    <extension name="Javahelp" href="https://intelunit2.vc.bravenet.com/kvm/VI/javahelp.php"></extension>
line: 48    <j2se version="1.6+"></j2se>
line: 47  <resources>
line: 46  </resources>
line: 45    <nativelib href="Linux_x86_64.jar"></nativelib>
line: 44    <j2se version="1.6+"></j2se>
line: 43  <resources os="Linux" arch="amd64">
line: 42  </resources>   
line: 41    <nativelib href="Linux_x86_64.jar"></nativelib>
line: 40    <j2se version="1.6+"></j2se>
line: 39  <resources os="Linux" arch="x86_64">
line: 38  </resources>
line: 37    <nativelib href="Linux_x86_32.jar"></nativelib>
line: 36    <j2se version="1.6+"></j2se>
line: 35  <resources os="Linux" arch="i386">
line: 34  </resources>
line: 33    <nativelib href="Linux_x86_32.jar"></nativelib>
line: 32    <j2se version="1.6+"></j2se>
line: 31  <resources os="Linux" arch="x86">
line: 30  </resources>
line: 29    <nativelib href="Win64.jar"></nativelib>
line: 28    <j2se version="1.6+"></j2se>
line: 27  <resources os="Windows" arch="amd64">
line: 26  </resources>
line: 25    <nativelib href="Win32.jar"></nativelib>
line: 24    <j2se version="1.6+"></j2se>
line: 23    <resources os="Windows" arch="x86">
line: 22  </resources>
line: 21    <jar href="https://intelunit2.vc.bravenet.com/kvm/VI/JViewer.jar" version="3.30a" download="eager" main="false"></jar>
line: 20    <j2se version="1.6+"></j2se>
line: 19  <resources>
line: 18  </security>
line: 17    <all-permissions></all-permissions>
line: 16  <security>
line: 15  </information>
line: 14    <description kind="tooltip">JViewer Networked KVM Application</description>
line: 13    <description kind="one-line">JViewer Networked KVM Application</description>
line: 12        </description>
line: 11            the remote computer to give him complete control over the remote machine.
line: 10            It also enables the user to redirect his local keyboard, mouse, CD-ROM and floppy drives to
line: 9            JViewer enables a user to view the video display of managed server via KVM.
line: 8    <description kind="short">
line: 7    <homepage href="null"></homepage>
line: 6    <vendor>Intel Corporation</vendor>
line: 5    <title>Networked KVM applet for Intel(R) Modular Server</title>
line: 4  <information>
line: 3 <jnlp spec="1.0+" codebase="https://intelunit2.vc.bravenet.com/kvm/VI/" xmlns="http://www.w3.org/1999/xhtml">
line: 2
<?xml version="1.0" standalone="yes"?>
Using MalformedXMLParser
JNLP file location: KvmViewer_3.30a-1.jnlp
Proceeding with jnlp
Read 577 entries from Firefox's preferences
Found preferences file: /home/ncoulson/.mozilla/firefox/scymb86a.default/prefs.js
Using firefox's profiles file: /home/ncoulson/.mozilla/firefox/profiles.ini
Starting security dialog thread
Forcing of offline set to: false
WARNING: key deployment.system.cachedir has no value, setting to default value
Loading User level properties from: file:/home/ncoulson/.config/icedtea-web/deployment.properties
Ico provider registered correctly.
[KvmViewer_3.30a-1.jnlp.txt](https://github.com/AdoptOpenJDK/IcedTea-Web/files/3740494/KvmViewer_3.30a-1.jnlp.txt)


cache: /home/ncoulson/.cache/icedtea-web file exists:true
config: /home/ncoulson/.config/icedtea-web file exists: true
System is already following XDG .cache and .config specifications

--- a/netx/net/sourceforge/nanoxml/XMLElement.java
+++ b/netx/net/sourceforge/nanoxml/XMLElement.java
@@ -1002,7 +1002,8 @@ public class XMLElement {
         }
         if (buf.length() == 0) {
             while (ch != '/') {
-                if (ch == '!') {
+                // jnlp files from a MFSys25  contain '<?-- JViewerVersion 3.30a -->', Treat as a comment
+                if (ch == '!' || ch == '?') {
                     ch = this.readChar();
                     if (ch != '-') {
                         throw this.expectedInput("Comment or Element");

